### PR TITLE
Remove margin if no input text

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?
This takes away the white space to the left of the Sign Up button in the case where the input text is hidden.

## Why?
It was [picked up](https://github.com/guardian/dotcom-rendering/pull/10198#issuecomment-1892427078) on the [PR that introduced it](https://github.com/guardian/dotcom-rendering/pull/10198)  

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| daily | email-present||

The component when the input is present:
no-email

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
Very clean!

I think a lot of the 19 commits can be removed by rebasing this branch?

**Review by mxdvl — COMMENTED** [review]
Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…

**Comment by lindseydew** [comment]
> Very clean!
> 
> I think a lot of the 19 commits can be removed by rebasing this branch?

Yep, we can squash those

**Comment by lindseydew** [comment]
> Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…

Yeah we took this approach initially. The problem was that the input box was displayed by default and then removed if this email address was there after the async call and it looked jumpy on the page

**Comment by lindseydew** [comment]
> > Just a thought: we could use the `gap` property on the flex parent and simplay `display: none` on the input if `signedInUserEmail` is truthy…
> 
> ~Yeah we took this approach initially. The problem was that the input box was displayed by default and then removed if this email address was there after the async call and it looked jumpy on the page~

Think I'd misunderstood. 

I added a gap to the parent element, and removed the conditional margin-right on the input box. However, this does still introduce a space before the button in the case where the input box is hidden. I think because the button is still showing in the second flex position? Please let me know if I am missing something though

```
const flexParentStyles = css`
	display: flex;
	flex-direction: row;
	align-items: flex-start;
	flex-wrap: wrap;
	gap: ${space[3]}px;
`;
```

**Comment by mxdvl** [comment]
> I added a gap to the parent element, and removed the conditional margin-right on the input box. However, this does still introduce a space before the button in the case where the input box is hidden. I think because the button is still showing in the second flex position?

Yes, for this approach to work we would need to ensure that the text input containter `div`  is set to `display: none;` whenever there’s a user email, rather than setting its `flex-basis` to 0.

File: dotcom-rendering/src/components/SecureSignup.importable.tsx [hunk 1]
```diff
@@ -343,41 +344,30 @@ export const SecureSignup = ({ newsletterId, successDescription }: Props) => {
...
-								? 'inline-block'
-								: 'none'};
-						`,
-					]}
+					cssOverrides={[labelStyles]}
+					style={
+						signedInUserEmail
+							? { display: 'none' }
+							: { display: 'inline-block' }
+					}
```
Comment (mxdvl) [inline dotcom-rendering/src/components/SecureSignup.importable.tsx:352]: You can also set the value to `undefined` to let it just be the default value: 

```suggestion
					style={{
						display:
						signedInUserEmail ? 'none' : undefined
					}}
```

**Review by mxdvl — APPROVED** [review]
Great! Even more approved!

**Pull request merged**